src/NMM_Gateway.php

```diff
@@ -339,23 +339,23 @@ private function get_qr_prefix($crypto) {
         return strtolower(str_replace(' ', '', $crypto->get_name()));
     }
 
-    private function get_qr_code($crypto, $walletAddress, $cryptoTotal) {        
+    private function get_qr_code($crypto, $walletAddress, $cryptoTotal, $orderId) {        
         $dirWrite = NMM_ABS_PATH . '/assets/img/';
 
         $formattedName = $this->get_qr_prefix($crypto);
 
         $qrData = $formattedName . ':' . $walletAddress . '?amount=' . $cryptoTotal;
 
         try {
-            QRcode::png($qrData, $dirWrite . 'tmp_qrcode.png', QR_ECLEVEL_H);            
+            QRcode::png($qrData, $dirWrite . 'tmp' . $orderId . '_qrcode.png', QR_ECLEVEL_H);            
         }
         catch (\Exception $e) {
             NMM_Util::log(__FILE__, __LINE__, 'QR code generation failed, falling back...');
             $endpoint = 'https://api.qrserver.com/v1/create-qr-code/?data=';
             return $endpoint . $qrData;
         }
         $dirRead = NMM_PLUGIN_DIR . '/assets/img/';
-        return $dirRead . 'tmp_qrcode.png';
+        return $dirRead . 'tmp' . $orderId . '_qrcode.png';
```

Comment:
not so nice to construct the filename twice, put

        $fileName = 'tmp_qrcode_' . $orderId . '.png';

above and use $filename instead.